compiler/infra/ILWalk.hpp

```diff
@@ -60,15 +59,17 @@ class TreeTopIteratorImpl
 
    // For logging
    //
+   TR::Compilation  *_comp;
    const char       *_name; // NULL means we don't want logging
-   TR::Optimization *_opt;
+
+   TR::Compilation * comp() { return _comp; }
 
    void logCurrentLocation();
 
    void stepForward();
    void stepBackward();
 
-   TreeTopIteratorImpl(TR::TreeTop *start, TR::Optimization *opt=NULL, const char *name=NULL);
+   TreeTopIteratorImpl(TR::TreeTop *start, TR::Compilation *comp=NULL, const char *name=NULL);
```

Comment:
I think the parameter should be mandatory too.  I don't quite understand your "scope" comment though because to fix this is simply a matter of removing the default value.  If there are places in the code that rely on the default value and don't pass in a valid compilation object I would argue that they are now potentially incorrect anyway because some of the underlying validator code relies on having a compilation object (whose access is not always guarded).